tests/test_tarball_generation.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+import time
+import requests
+import signal
+import tarfile
+import shutil
+
+# 1. Extract the tarball
+tarball_path = "target/unitycatalog-0.1.0.tar.gz"
+extract_path = "target/dist"
+
+if not os.path.exists(extract_path):
+    os.makedirs(extract_path)
+
+with tarfile.open(tarball_path, "r:gz") as tar:
+    tar.extractall(path=extract_path, filter='data')
+    print("Extracted tarball to {}".format(extract_path))
+
+# 2. Check executable permissions
+bin_dir = extract_path + "/bin"
+for filename in os.listdir(bin_dir):
+    filepath = os.path.join(bin_dir, filename)
+    if os.path.isfile(filepath):
+        os.chmod(filepath, 0o755)
+        print(f"Set executable permission for {filepath}")
+
+# 3. Start server script
+start_server_cmd = os.path.join(bin_dir, "start-uc-server")
+server_process = subprocess.Popen([start_server_cmd], stdout=subprocess.PIPE, stderr=subprocess.PIPE, preexec_fn=os.setsid)
+with open("server_pid.txt", "w") as f:
+    f.write(str(server_process.pid))
+print(f"Server started with PID {server_process.pid}")
+time.sleep(10)  # Give the server some time to start
```

Comment:
this may be flaky, make it 60. no harm. ideally, its best to check repeatedly in a loop so that we dont wait for too long.. but thats okay, this is not the most time-consuming test